code/modules/surgery/implant_removal.dm

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////
+//					IMPLANT REMOVAL SURGERY						//
+//////////////////////////////////////////////////////////////////
+
+/datum/surgery/implant_removal
+	name = "Implant Removal"
+	steps = list(/datum/surgery_step/generic/cut_open, /datum/surgery_step/generic/clamp_bleeders, /datum/surgery_step/generic/retract_skin,/datum/surgery_step/extract_implant,/datum/surgery_step/generic/cauterize)
+	possible_locs = list("chest")
+
+/datum/surgery/implant_removal/synth
+	name = "Implant Removal"
+	steps = list(/datum/surgery_step/robotics/external/unscrew_hatch,/datum/surgery_step/robotics/external/open_hatch,/datum/surgery_step/extract_implant,/datum/surgery_step/robotics/external/close_hatch)
+	possible_locs = list("chest")
+
+/datum/surgery/implant_removal/can_start(mob/user, mob/living/carbon/human/target)
+	if(!istype(target))
+		return 0
+	var/obj/item/organ/external/affected = target.get_organ(user.zone_sel.selecting)
+	if(!affected)
+		return 0
+	if(affected.status & ORGAN_ROBOT)
+		return 0
+	return 1
+
+/datum/surgery/implant_removal/synth/can_start(mob/user, mob/living/carbon/human/target)
+	if(!istype(target))
+		return 0
+	var/obj/item/organ/external/affected = target.get_organ(user.zone_sel.selecting)
+	if(!affected)
+		return 0
+	if(!(affected.status & ORGAN_ROBOT))
+		return 0
+
+	return 1
+
+/datum/surgery_step/extract_implant
+	name = "extract implant"
+	allowed_tools = list(/obj/item/weapon/hemostat = 100, /obj/item/weapon/crowbar = 65)
+	time = 64
+	var/obj/item/weapon/implant/I = null
+
+/datum/surgery_step/extract_implant/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	I = locate(/obj/item/weapon/implant) in target
```

Comment:
This is just moving around the code; not going to bother fixing any but the most major/glaring of issues.